src/dotnet/ToolPackage/IToolPackageStore.cs

```diff
@@ -4,13 +4,22 @@
 using System;
 using System.Collections.Generic;
 using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Packaging.Core;
 
 namespace Microsoft.DotNet.ToolPackage
 {
     internal interface IToolPackageStore
     {
         DirectoryPath Root { get; }
 
-        IEnumerable<IToolPackage> GetInstalledPackages(string packageId = null);
+        DirectoryPath GetRandomStagingDirectory();
```

Comment:
If ToolPackageStore only used by ToolPackageInstaller it would be fine with this. In this way, InstallToolCommand would also know that. And also, since it is a GetRandomStagingDirectory, why this logic need to be in that interface?

Reply:
It's also used by uninstall so that we can move packages into staging prior to permanently deleting them; we can't move to `temp` because that's likely to be a cross-mount move and thus non-transactional (Sysetm.IO.Directory throws).